client/main.lua

```diff
@@ -42,37 +35,33 @@ end)
 
 AddEventHandler('onResourceStart', function(resourceName)
     if GetCurrentResourceName() ~= resourceName or not isLoggedIn then return end
-    PlayerData = QBCore.Functions.GetPlayerData()
-    hasFitbit = fitbitCheck(PlayerData.items)
+    hasFitbit = fitbitCheck(QBX.PlayerData.items)
 end)
 
 RegisterNetEvent('QBCore:Client:OnPlayerLoaded', function()
-    PlayerData = QBCore.Functions.GetPlayerData()
-    hasFitbit = fitbitCheck(PlayerData.items)
+    hasFitbit = fitbitCheck(QBX.PlayerData.items)
 end)
 
 RegisterNetEvent('QBCore:Client:OnPlayerUnload', function()
     hasFitbit = false
-    PlayerData = {}
 end)
 
-RegisterNetEvent('QBCore:Player:SetPlayerData', function(val)
-    PlayerData = val
-    hasFitbit = fitbitCheck(PlayerData.items)
+RegisterNetEvent('QBCore:Player:SetPlayerData', function()
+    hasFitbit = fitbitCheck(QBX.PlayerData.items)
```

Comment:
Use the ox_inventory event to check for existence of an item

Reply:
Hopefully the new commit resolves this